src/js/vue.js

```diff
@@ -1,9 +1,11 @@
 import route from './index.js';
 
 export const ZiggyVue = {
-    install: (v, options) => v.mixin({
-        methods: {
-            route: (name, params, absolute, config = options) => route(name, params, absolute, config),
-        },
-    }),
+    install: (v, options) => {
+        const version = parseInt(v.version.split('.')[0]);
+        const _route = (name, params, absolute, config = options) => route(name, params, absolute, config);
+
+        v.mixin && version === 2 && (v.mixin({ methods: { route: _route } }));
+        v.provide && version > 2 && (v.provide('route', _route));
```

Comment:
Why the `v.provide` check? If we're on Vue 3 `provide` is always going to be there, right?

Reply:
You're right, that can be removed, probably overkill.